core/src/main/java/team/techtigers/core/RobotSaveState.java

```diff
@@ -0,0 +1,58 @@
+package team.techtigers.core;
+
+import java.util.HashMap;
+
+/**
+ * Saves any value from one opmode to be used in another
+ */
+public class RobotSaveState {
+    private HashMap<String, Object> state;
+    private static RobotSaveState instance;
+
+    private RobotSaveState(){
+        state = new HashMap<>();
+    }
+
+    /**
+     * If there is an instance of the class, returns it
+     * Otherwise, creates a new instance and returns it
+     * @return The instance of the class
+     */
+    public static RobotSaveState getInstance() {
+        if (instance == null) {
+            instance = new RobotSaveState();
+        }
+
+        return instance;
+    }
+
+    /**
+     * Gets the current robot state
+     * @return The current robot state
+     */
+    public HashMap<String, Object> getState() {
```

Comment:
So this should take an input of a String, and the return should be something like state.get(key). We aren't returning the whole state, just a specific value requested

Reply:
Should have thought of that, thanks for clarifying